content/en/blog/_posts/2021-03-09-Defining-NetworkPolicy-Conformance-For-CNIs.md

```diff
@@ -0,0 +1,477 @@
+
+- [Antrea: NetworkPolicy: unable to allow ingress by CIDR](https://github.com/vmware-tanzu/antrea/issues/1764)
+- [Calico: default missing protocol to TCP; don't let single port overwrite all ports](https://github.com/projectcalico/libcalico-go/pull/1373)
+- [Cilium: Egress Network Policy allows traffic that should be denied](https://github.com/cilium/cilium/issues/14678)
+
+The good news is, all of these issues are actively being fixed and iterated on between SIG Network,
+the CNI providers, and the Network Policy subproject.
+
+Are you interested in verifying NetworkPolicy functionality on your cluster?
+(if you care about security or offer multi-tenant SaaS, you should be)
+If so, you can run the upstream end to end tests, or Cyclonus, or both.
+- If you're just getting started with NetworkPolicies and want to simply
+  verify the "common" NetworkPolicy cases that most CNIs should be
+  implementing correctly, in a way that is quick to diagnose, then you're
+  better off running the e2e tests only.
+- If you are deeply curious about your CNI provider's NetworkPolicy
+  implementation, and want to verify it: use Cyclonus.
+- If you want to test *hundreds* of policies, and evaluate your CNI plugin
+  for comprehensive functionality, for deep discovery of potential security
+  holes: use Cyclonus, and also consider running end-to-end cluster tests.
+- If you're thinking of getting involved with the upstream NetworkPolicy efforts:
+  use Cyclonus, and read at least an outline of what e2e tests are relevant.
+
+## Where to start with NetworkPolicy testing?
+
+- Cyclonus is easy to run on your cluster, check out the [instructions on github](https://github.com/mattfenwick/cyclonus#run-as-a-kubernetes-job),
+  and determine whether *your* specific CNI configuration is fully conformant to the hundreds of different
+  Kubernetes Network Policy API constructs.
+- Alternatively, you can use a tool like [sonobuoy](https://github.com/vmware-tanzu/sonobuoy)
+  to run the existing E2E tests in Kubernetes, with the `--ginkgo.focus=NetworkPolicy` flag.
+  Note that only versions of sonobuoy built for k8s 1.21 and above will have the *new* Network Policy tests in them.
+
+## Improvements to the NetworkPolicy API and user experience
+
+In addition to cleaning up the validation story for CNI plugins that implement NetworkPolicies,
+we've also spent some time improving the Kubernetes NetworkPolicy API for a few commonly requested features.
+After months of deliberation, we eventually settled on a few core areas for improvement:
+
+- Port Range policies: We now allow you to specify a *range* of ports for a policy.
+  This allows users interested in scenarios like FTP or virtualization to enable advanced policies.
+  The port range option for network policies will be available to use in Kubernetes 1.21.
+  You can read more about this [here](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2079-network-policy-port-range).
```

Comment:
```suggestion
  Read more in [targeting a range of ports](/docs/concepts/services-networking/network-policies/#targeting-a-range-of-ports).
```

Reply:
oops accepting this change cleared the lgtm label 😱